__tests__/app.test.js

```diff
@@ -164,5 +159,253 @@ describe("post/api/articles/:articleid/comments", () => {
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            article_img_url: expect.any(String),
+            votes: expect.any(Number),
+            author: expect.any(String),
+            body: expect.any(String),
+          })
+        );
+        expect(body.votes).toBe(10);
+      });
+  });
+  test("returns 404 error if article id that isnt in database is given ", () => {
+    const votes = { inc_votes: 10 };
+    return request(app)
+      .patch("/api/articles/5000")
+      .send(votes)
+      .expect(404)
+      .then((body) => {
+        expect(body._body.msg).toBe("Invalid article ID or no votes passed");
+      });
+  });
+  test("returns 404 error if invalid format for votes given", () => {
+    const votes = { votes: "10" };
+    return request(app)
+      .patch("/api/articles/3")
+      .send(votes)
+      .expect(404)
+      .then((body) => {
+        expect(body._body.msg).toBe("Invalid data type");
+      });
+  });
+});
 
+describe("get/api/users", () => {
+  test("Return users with the correct properties with 200", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toHaveLength(4);
+        body.forEach((user) => {
+          expect(user).toEqual(
+            expect.objectContaining({
+              username: expect.any(String),
+              name: expect.any(String),
+              avatar_url: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+});
+
+describe("api queries", () => {
+  test("returns correct topic to the topic query", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(11);
+        body.articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              title: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              article_img_url: expect.any(String),
+              article_id: expect.any(Number),
+              votes: expect.any(Number),
+              author: expect.any(String),
+              comment_count: expect.any(String),
+            })
+          );
+        });
+      });
+  });
 
+  test("returns correct order when using sort_by with order", () => {
+    return request(app)
+      .get("/api/articles?sort_by=topic&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("topic", { ascending: true });
+      });
+  });
+  test("combines all 3 queries", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch&sort_by=author&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("author", { ascending: true });
+      });
+  });
+  test("invalid query given returns data with default queries", () => {
```

Comment:
11: Please also add a test to check that the defaults work without any queries. e.g. just /api/articles should be sorted by created_at